core/database_util.go

```diff
@@ -164,6 +95,20 @@ func GetHeadFastBlockHash(db ethdb.Database) common.Hash {
 	return common.BytesToHash(data)
 }
 
+// GetHeadBlockNum retrieves the block number of the current canonical head block.
+func GetHeadBlockNum(db ethdb.Database) *big.Int {
```

Comment:
I don't think this method is a good idea as it's a very expensive database operation just to get the head number. The reason no such methods were written is that when going through the blockchain (i.e. chain.GetXxx), all intermediate results that are pulled from the database are cached, or pulled from the cache. This method will hide the expensive db op behind an innocent name and will incur significant runtime costs.